akka-persistence-typed/src/test/java/akka/persistence/typed/javadsl/PersistentActorJavaDslTest.java

```diff
@@ -131,11 +133,19 @@ public int hashCode() {
     final int value;
     final List<Integer> history;
 
+    static final State EMPTY = new State(0, Collections.emptyList());
+
     public State(int value, List<Integer> history) {
       this.value = value;
       this.history = history;
     }
 
+    public State incrementedBy(int delta) {
```

Comment:
In general, I think we should recommend implementing state transformations in the state classes. It's much easier to unit test and IMHO more clear.